File: 프로그래머스/1/67256. ［카카오 인턴］ 키패드 누르기/［카카오 인턴］ 키패드 누르기.java
class Solution {
    public String solution(int[] numbers, String hand) {
        String answer = "";
        //처음 위치 left:*(10),right:#(11)
        int leftHand = 10;
        int rightHand = 11;
        int[][] coordinate = {{0, 3},{-1, 0},{0, 0},{1, 0},{-1, 1},{0, 1},{1, 1},{-1, 2},{0, 2},{1, 2},{-1,3},{1,3}};
        
        for(int i=0; i<numbers.length; i++){
            if(numbers[i] == 1 || numbers[i] == 4 || numbers[i] == 7){
                answer +="L";
                leftHand = numbers[i];
            }else if(numbers[i] == 3 || numbers[i] == 6 || numbers[i] == 9){
                answer +="R";
                rightHand = numbers[i];
            }else{
                int leftSum = Math.abs(coordinate[numbers[i]][0] - coordinate[leftHand][0]) + Math.abs(coordinate[numbers[i]][1] - coordinate[leftHand][1]);
                int rightSum = Math.abs(coordinate[numbers[i]][0] - coordinate[rightHand][0]) + Math.abs(coordinate[numbers[i]][1] - coordinate[rightHand][1]);
                
                
               if(leftSum > rightSum){
                   answer +="R";
                   rightHand = numbers[i];
               }else if(leftSum < rightSum){
                   answer +="L";
                   leftHand = numbers[i];
               }else{
                   if(hand.equals("right")){
                       answer +="R";
                       rightHand = numbers[i];
                   }else{
                       answer +="L";
                       leftHand = numbers[i];
                   };
               }
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/67256. ［카카오 인턴］ 키패드 누르기/［카카오 인턴］ 키패드 누르기.py
def solution(numbers, hand):
    answer = ''
    xy = [[1,3],[0,0],[1,0],[2,0],[0,1],[1,1],[2,1],[0,2],[1,2],[2,2]]
    positionL = [0,3]
    positionR = [2,3]
    
    for n in numbers:
        if n == 1 or n == 4 or n == 7:
            positionL = xy[n][0],xy[n][1]
            answer+="L"
        elif n == 3 or n == 6 or n == 9:
            positionR = xy[n][0],xy[n][1]
            answer+="R"
        elif n == 2 or n == 5 or n == 8 or n == 0:
            a = abs(positionL[0] - xy[n][0])+abs(positionL[1] - xy[n][1])
            b = abs(positionR[0] - xy[n][0])+abs(positionR[1] - xy[n][1])
            if a > b:
                positionR = xy[n][0],xy[n][1]
                answer+="R"
            elif a < b:
                positionL = xy[n][0],xy[n][1]
                answer+="L"
            else:
                if hand == "right":
                    positionR = xy[n][0],xy[n][1]
                    answer+="R"
                else:
                    positionL = xy[n][0],xy[n][1]
                    answer+="L"
                    
    return answer